### PR TITLE
Update manifest file schema

### PR DESCRIPTION
**Pull request by mbiannaccone**
This PR updates the manifest file schema to make tags a part of the schema itself. 

still todo:
- [x] make the tag check a warning, not an error

### CODE REVIEW

File: canvas_cli/templates/plugins/default/{{ cookiecutter.project_slug }}/README.md [hunk 1]
```diff
@@ -8,11 +8,14 @@
 
 ```
 {{ cookiecutter.project_slug }}/
-├── commands/
-├── content/
-├── effects/
-├── protocols/
-├── views/
+├── {{ cookiecutter.project_slug }}/
```
Comment (aduane) [inline canvas_cli/templates/plugins/default/{{ cookiecutter.project_slug }}/README.md:11]: We don't want this extra nesting, the CANVAS_PLUGIN_TAGS, or the pyproject.toml

File: canvas_cli/utils/validators/manifest_schema.py [hunk 2]
```diff
@@ -17,8 +17,43 @@
...
+        "plugin_version",
+        "name",
+        "description",
+        "components",
+        "tags",
+        "references",
+        "license",
+        "diagram",
+        "readme",
+        "public",
```
Comment (aduane) [inline canvas_cli/utils/validators/manifest_schema.py:55]: Public won't be set in the manifest.

File: canvas_cli/utils/validators/manifest_schema.py [hunk 3]
```diff
@@ -17,8 +17,43 @@
...
+    "required": [
+        "sdk_version",
+        "plugin_version",
+        "name",
+        "description",
+        "components",
+        "tags",
+        "references",
+        "license",
+        "diagram",
```
Comment (aduane) [inline canvas_cli/utils/validators/manifest_schema.py:53]: diagram not trequired

File: canvas_cli/utils/validators/manifest_schema.py [hunk 4]
```diff
@@ -17,8 +17,43 @@
...
     },
-    "required": ["sdk_version", "plugin_version", "name", "description", "components"],
+    "required": [
+        "sdk_version",
+        "plugin_version",
+        "name",
+        "description",
+        "components",
+        "tags",
+        "references",
```
Comment (aduane) [inline canvas_cli/utils/validators/manifest_schema.py:51]: Can these be empty lists? If not, I don't think they should be required.

File: canvas_cli/templates/plugins/default/{{ cookiecutter.project_slug }}/README.md (hunk 1 above)
Comment (mbiannaccone) [reply canvas_cli/templates/plugins/default/{{ cookiecutter.project_slug }}/README.md:11]: i think @beaugunderson and @jrwils  had to add back in the pyproject.toml and the nesting for the install to work. i'll let them weigh in on why!

Comment (beaugunderson) [reply canvas_cli/templates/plugins/default/{{ cookiecutter.project_slug }}/README.md:11]: we did for the demo but yeah we want to rip that all out

Comment (mbiannaccone) [reply canvas_cli/templates/plugins/default/{{ cookiecutter.project_slug }}/README.md:11]: ok good to know, i can make that change!

**Pull request merged**